File: 2425-Bitwise XOR of All Pairings.py
class Solution(object):
    def xorAllNums(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: int
        """
        remainder1, remainder2 = len(nums1)%2, len(nums2)%2

        # if not remainder1 and not remainder2:
        #     return 0
        # elif remainder1 and not remainder2:
        #     return reduce(lambda x, y: x^y, nums2, 0)
        # elif remainder2 and not remainder1:
        #     return reduce(lambda x, y: x^y, nums1, 0)
        # return reduce(lambda x, y: x^y, nums2, 0)^reduce(lambda x, y: x^y, nums1, 0)
    
# more concise:

        # return (remainder1*reduce(lambda x, y: x^y, nums2, 0))^(remainder2*reduce(lambda x, y: x^y, nums1, 0))

# even more concise:
        return (remainder1*reduce(xor, nums2))^(remainder2*reduce(xor, nums1))

File: C++/2425-Bitwise XOR of All Pairings.cpp
class Solution {
public:
    int xorAllNums(vector<int>& nums1, vector<int>& nums2) {
        int N1 = nums1.size(), N2 = nums2.size();
        int res = 0;
        if (N2%2) {
            for (int num:nums1) {
                res ^= num;
            }
        }
        if (N1%2) {
            for (int num:nums2) {
                res ^= num;
            }
        }
        return res;
    }
};